internal/provider/resource_locally_signed_cert_test.go

```diff
@@ -5,139 +5,132 @@ import (
-					if err != nil {
-						return fmt.Errorf("error parsing cert: %s", err)
-					}
-					if expected, got := "2", cert.Subject.SerialNumber; got != expected {
-						return fmt.Errorf("incorrect subject serial number: expected %v, got %v", expected, got)
-					}
-					if expected, got := "example.com", cert.Subject.CommonName; got != expected {
-						return fmt.Errorf("incorrect subject common name: expected %v, got %v", expected, got)
-					}
-					if expected, got := "Example, Inc", cert.Subject.Organization[0]; got != expected {
-						return fmt.Errorf("incorrect subject organization: expected %v, got %v", expected, got)
-					}
-					if expected, got := "Department of Terraform Testing", cert.Subject.OrganizationalUnit[0]; got != expected {
-						return fmt.Errorf("incorrect subject organizational unit: expected %v, got %v", expected, got)
-					}
-					if expected, got := "5879 Cotton Link", cert.Subject.StreetAddress[0]; got != expected {
-						return fmt.Errorf("incorrect subject street address: expected %v, got %v", expected, got)
-					}
-					if expected, got := "Pirate Harbor", cert.Subject.Locality[0]; got != expected {
-						return fmt.Errorf("incorrect subject locality: expected %v, got %v", expected, got)
-					}
-					if expected, got := "CA", cert.Subject.Province[0]; got != expected {
-						return fmt.Errorf("incorrect subject province: expected %v, got %v", expected, got)
-					}
-					if expected, got := "US", cert.Subject.Country[0]; got != expected {
-						return fmt.Errorf("incorrect subject country: expected %v, got %v", expected, got)
-					}
-					if expected, got := "95559-1227", cert.Subject.PostalCode[0]; got != expected {
-						return fmt.Errorf("incorrect subject postal code: expected %v, got %v", expected, got)
-					}
-
-					if expected, got := 2, len(cert.DNSNames); got != expected {
-						return fmt.Errorf("incorrect number of DNS names: expected %v, got %v", expected, got)
-					}
-					if expected, got := "example.com", cert.DNSNames[0]; got != expected {
-						return fmt.Errorf("incorrect DNS name 0: expected %v, got %v", expected, got)
-					}
-					if expected, got := "example.net", cert.DNSNames[1]; got != expected {
-						return fmt.Errorf("incorrect DNS name 0: expected %v, got %v", expected, got)
-					}
+				Config: locallySignedCertConfig(1, 0),
+				Check: r.ComposeAggregateTestCheckFunc(
+					r.TestMatchResourceAttr("tls_locally_signed_cert.test", "cert_pem", regexp.MustCompile(`^-----BEGIN CERTIFICATE----`)),
+					r.TestCheckResourceAttrWith("tls_locally_signed_cert.test", "cert_pem", func(value string) error {
+						block, _ := pem.Decode([]byte(value))
+						cert, err := x509.ParseCertificate(block.Bytes)
+						if err != nil {
+							return fmt.Errorf("error parsing cert: %s", err)
+						}
+						if expected, got := "2", cert.Subject.SerialNumber; got != expected {
+							return fmt.Errorf("incorrect subject serial number: expected %v, got %v", expected, got)
+						}
```

Comment:
Nit: Similar question for creating a provider codebase test helper for `x509.Certificate`